client/sites-dashboard-v2/hooks/use-initialize-dataviews-page.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useRef } from 'react';
+import type { DataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews/interfaces';
+
+// DataViews' pagination always resets when the search component is mounted, even though the search term has not changed.
+// This is a bug which has a fix in https://github.com/WordPress/gutenberg/pull/61307.
+// This is a workaround until the above fix is released.
+// Here, we restore the page to the previous page if it is unintentionally changed by the above bug.
+export function useInitializeDataViewsPage(
+	dataViewsState: DataViewsState,
+	setDataViewsState: ( state: DataViewsState ) => void
+) {
+	const prevPage = useRef( dataViewsState.page );
+	const prevSearch = useRef( dataViewsState.search );
+
+	const done = useRef( false );
+
+	useEffect( () => {
+		if ( prevPage.current === 1 ) {
+			done.current = true;
+		}
+		if ( done.current ) {
+			return;
+		}
+
+		if (
+			dataViewsState.search === prevSearch.current &&
+			dataViewsState.page !== prevPage.current
```

Comment:
Currently (outside of this PR) our filtering isnt initialized correctly on load. That should be fixed simply with https://github.com/Automattic/wp-calypso/pull/90749 

Im wondering once that is working if we are going to need to check `dataViewsState.filter === prevFilter.current` type thing here as well?

Reply:
Thanks for checking! Let's merge the above PR and then we can rebase this PR and see if anything breaks. 😄

Reply:
Sounds good, just merged it! 👍  Thanks for the quick review 😁

Reply:
Rebased and tested and it seems to still work correctly.

Note that pagination will reset to 1 when the status filter is changed. I think it is as per DataViews' behavior.

Reply:
> Note that pagination will reset to 1 when the status filter is changed

I think that is fine and expected. If we change the list pagination should reset, I don't think preserving it would have any meaning unless we were trying to implement some current-page-only type filtering.